drivers/stepper/adi_tmc/adi_tmc5xxx_common.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file drivers/stepper/adi_tmc/adi_tmc5xxx_common.h
+ *
+ * @brief Common TMC5xxx stepper controller driver definitions
+ */
+
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Jilay Sandeep Pandya
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_STEPPER_ADI_TMC_ADI_TMC5XXX_COMMON_H_
+#define ZEPHYR_DRIVERS_STEPPER_ADI_TMC_ADI_TMC5XXX_COMMON_H_
+
+#include "adi_tmc_reg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name TMC5xxx module functions
+ * @anchor TMC5XXX_FUNCTIONS
+ *
+ * @{
+ */
+
+/**
+ * @brief Calculate the velocity in full clock cycles from the velocity in Hz
+ *
+ * @param velocity_hz Velocity in Hz
+ * @param velocity_fclk Pointer to store the velocity in full clock cycles
+ * @param clock_frequency Clock frequency in Hz
+ */
+inline void tmc5xxx_calculate_velocity_from_hz_to_fclk(const uint32_t velocity_hz,
+						       uint32_t *velocity_fclk,
+						       const uint32_t clock_frequency)
```

Comment:
I would update the signature to:
```
static inline uint32_t tmc5xxx_calculate_velocity_from_hz_to_fclk(uint64_t velocity_hz,
						                  uint32_t clock_frequency)
{
        ASSERT_NO_MSG(clock_frequency);
        return (velocity_hz << TMC5XXX_CLOCK_FREQ_SHIFT) / clock_frequency;
}
```
* static inline to allow the compiler to either inline it or make it a static function, whichever is most efficient.
* since the function "can't fail", just return the result
* add an assert to prevent divide by zero error
* make `velocity_hz` 64 bit removing the need to promote it and make it mutable (no const) to allow the compiler to work directly on the variable which is already on the stack
* since we are copying the entire data type, const in the function really does not help us :) const is useful when referencing a structure outside of the function, so we don't modify something we don't own, inside the function its not really helpful :)